### ISSUE TITLE
E_INVALIDARG - One or more arguments are invalid

### ISSUE THREAD

**Issue by qgrosperrin** [ly4k/Certipy#116]
Hi. Great tool, thanks for your work !

I am having a similar issue as https://github.com/ly4k/Certipy/issues/81 but I have verified that my CA name is correct, as well as the target (as suggested in the issue).

It should be noted we're using Kerberos, so I don't know if this has any impact. Here is the full sanitised output below
```
proxychains4 certipy req -dc-ip <dc-ip> -u username@DOMAIN.NET -k -no-pass -dns-tcp -target CA1.DOMAIN.NET -target-ip <CA-ip>  -ca 'CA1.DOMAIN.NET' -template '<template-name>' -upn 'target-user@DOMAIN.NET' -debug
Certipy v4.0.0 - by Oliver Lyak (ly4k)

[+] Domain retrieved from CCache: DOMAIN.NET
[+] Username retrieved from CCache: username
[+] Generating RSA key
[*] Requesting certificate via RPC
[+] Using Kerberos Cache: /media/sf_tools/username@DOMAIN.NET.ccache
[+] Using TGT from cache
[+] Username retrieved from CCache: username
[+] Getting TGS for 'host/CA1.DOMAIN.NET'
[proxychains] Strict chain  ...  <socks-ip>:8064  ...  <dc-ip>:88  ...  OK
[+] Got TGS for 'host/CA1.DOMAIN.NET'
[+] Trying to connect to endpoint: ncacn_np:<CA-ip>[\pipe\cert]
[proxychains] Strict chain  ...  <socks-ip>:8064  ...  <CA-ip>:445  ...  OK
[!] Failed to connect to endpoint ncacn_np:<CA-ip>[\pipe\cert]: SMB SessionError: STATUS_ACCESS_DENIED({Access Denied} A process has requested access to an object but has not been granted those access rights.)
[+] Trying to resolve dynamic endpoint '91AE6020-9E3C-11CF-8D7C-00AA00C091BE'
[proxychains] Strict chain  ...  <socks-ip>:8064  ...  <CA-ip>:135  ...  OK
[+] Resolved dynamic endpoint '91AE6020-9E3C-11CF-8D7C-00AA00C091BE' to 'ncacn_ip_tcp:<CA-ip>[50894]'
[+] Using Kerberos Cache: /media/username@DOMAIN.NET.ccache
[+] Using TGT from cache
[+] Username retrieved from CCache: username
[+] Getting TGS for 'host/CA1.DOMAIN.NET'
[proxychains] Strict chain  ...  <socks-ip>:8064  ...  <dc-ip>:88  ...  OK
[+] Got TGS for 'host/CA1.DOMAIN.NET'
[+] Trying to connect to endpoint: ncacn_ip_tcp:<CA-ip>[50894]
[proxychains] Strict chain  ...  <socks-ip>:8064  ...  <CA-ip>:50894  ...  OK
[+] Connected to endpoint: ncacn_ip_tcp:<CA-ip>[50894]
[-] Got error: RequestSessionError: code: 0x80070057 - E_INVALIDARG - One or more arguments are invalid.
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/certipy/entry.py", line 60, in main
    actions[options.action](options)
  File "/usr/local/lib/python3.10/dist-packages/certipy/commands/parsers/req.py", line 12, in entry
    req.entry(options)
  File "/usr/local/lib/python3.10/dist-packages/certipy/commands/req.py", line 764, in entry
    request.request()
  File "/usr/local/lib/python3.10/dist-packages/certipy/commands/req.py", line 715, in request
    cert = self.interface.request(csr, attributes)
  File "/usr/local/lib/python3.10/dist-packages/certipy/commands/req.py", line 208, in request
    response = self.dce.request(request)
  File "/root/.local/lib/python3.10/site-packages/impacket/dcerpc/v5/rpcrt.py", line 880, in request
    raise exception
certipy.commands.req.DCERPCSessionError: RequestSessionError: code: 0x80070057 - E_INVALIDARG - One or more arguments are invalid.
```

**Comment by ly4k** [comment]
Hello. It seems that the "-ca" parameter is incorrect. It should _not_ be the host name of the CA server, but it should be the name of the Certificate Authority itself. The name of the CA can be found in the Certipy "find" output at the very top of the text output for instance. Can you please verify that? Perhaps send me the "certificate authorities" output from Certipy's find (in a redacted form if you prefer), and then I can show you what you should specify.

Best regards

**Comment by ly4k** [comment]
Thank you. Then it should be '-ca "Company Issuing CA 03"'.

Let me know if it works!

**Comment by qgrosperrin** [comment]
Thanks that fixed it !

**Comment by maseehabdul** [comment]
certipy-ad req -username 'svc_admin$' -password 'demo@123' -ca 'AUTHORITY-CA' -target 10.10.11.222 -template 'CorpVPN' -upn "administrator@authority.htb" -dns authority.authority.htb  -debug
Certipy v4.7.0 - by Oliver Lyak (ly4k)

[+] Trying to resolve '' at '183.82.243.66'
[+] Generating RSA key
[*] Requesting certificate via RPC
[+] Trying to connect to endpoint: ncacn_np:10.10.11.222[\pipe\cert]
[!] Failed to connect to endpoint ncacn_np:10.10.11.222[\pipe\cert]: SMB SessionError: STATUS_ACCESS_DENIED({Access Denied} A process has requested access to an object but has not been granted those access rights.)
[+] Trying to resolve dynamic endpoint '91AE6020-9E3C-11CF-8D7C-00AA00C091BE'
[+] Resolved dynamic endpoint '91AE6020-9E3C-11CF-8D7C-00AA00C091BE' to 'ncacn_ip_tcp:10.10.11.222[49716]'
[+] Trying to connect to endpoint: ncacn_ip_tcp:10.10.11.222[49716]
[+] Connected to endpoint: ncacn_ip_tcp:10.10.11.222[49716]
[-] Got error: Unknown DCE RPC fault status code: 00000721
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/certipy/entry.py", line 60, in main
    actions[options.action](options)
  File "/usr/lib/python3/dist-packages/certipy/commands/parsers/req.py", line 12, in entry
    req.entry(options)
  File "/usr/lib/python3/dist-packages/certipy/commands/req.py", line 767, in entry
    request.request()
  File "/usr/lib/python3/dist-packages/certipy/commands/req.py", line 718, in request
    cert = self.interface.request(csr, attributes)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/certipy/commands/req.py", line 208, in request
    response = self.dce.request(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/impacket/dcerpc/v5/rpcrt.py", line 859, in request
    answer = self.recv()
             ^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/impacket/dcerpc/v5/rpcrt.py", line 1332, in recv
    raise DCERPCException('Unknown DCE RPC fault status code: %.8x' % status_code)
impacket.dcerpc.v5.rpcrt.DCERPCException: Unknown DCE RPC fault status code: 00000721



this is my error
